apps/consumer-client/pages/popup.tsx

```diff
@@ -20,7 +20,14 @@ export default function AuthPopup() {
       window.location.href = decodeURIComponent(params.get("proofUrl")!);
     } else if (params.get("proof") != null) {
       // Later, the Passport redirects back with a proof. Send it to our parent.
-      window.opener.postMessage({ encodedPcd: params.get("proof")! }, "*");
+      window.opener.postMessage({ encodedPCD: params.get("proof")! }, "*");
+      window.close();
+    } else if (params.get("pendingStampPCD") != null) {
+      // Later, the Passport redirects back with a pendingStampPCD. Send it to our parent.
+      window.opener.postMessage(
+        { pendingStampPCD: params.get("pendingStampPCD")! },
```

Comment:
I generally try to avoid `!` - you can do this by extracting the variable and using that in both the `if` statement that checks if it's not `undefined` and the place that uses the variable within the `if` statement.